test/integration/projects/projects_test.go

```diff
@@ -0,0 +1,169 @@
+		},
+	} {
+		t.Run(tts.name, func(t *testing.T) {
+
+			shared := tft.NewTFBlueprintTest(t,
+				tft.WithTFDir(tts.tfDir),
+			)
+
+			shared.DefineApply(
+				func(assert *assert.Assertions) {
+					// perform default apply of the blueprint
+					shared.DefaultApply(assert)
+					// save the value of the "cloudbuild_sa" to be used in the envs tests
+					sharedData[tts.name].CloudbuildSA = shared.GetStringOutput("cloudbuild_sa")
+				})
+			shared.DefineVerify(
+				func(assert *assert.Assertions) {
+					// perform default verification ensuring Terraform reports no additional changes on an applied blueprint
+					shared.DefaultVerify(assert)
+					// save the value of the "cloudbuild_sa" to be used in the envs tests
+					sharedData[tts.name].CloudbuildSA = shared.GetStringOutput("cloudbuild_sa")
+				})
+			shared.Test()
+		})
+
+	}
+
+	for _, tt := range []struct {
+		name            string
+		sharedData      string
+		perimeterEnvVar string
+		tfDir           string
+		networkTfDir    string
+	}{
+		{
+			name:            "bu1_development",
+			sharedData:      "bu1_shared",
+			perimeterEnvVar: "TF_VAR_dev_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_1/development",
+			networkTfDir:    "../../../3-networks/envs/development",
+		},
+		{
+			name:            "bu1_non-production",
+			sharedData:      "bu1_shared",
+			perimeterEnvVar: "TF_VAR_nonprod_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_1/non-production",
+			networkTfDir:    "../../../3-networks/envs/non-production",
+		},
+		{
+			name:            "bu1_production",
+			sharedData:      "bu1_shared",
+			perimeterEnvVar: "TF_VAR_prod_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_1/production",
+			networkTfDir:    "../../../3-networks/envs/production",
+		},
+		{
+			name:            "bu2_development",
+			sharedData:      "bu2_shared",
+			perimeterEnvVar: "TF_VAR_dev_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_2/development",
+			networkTfDir:    "../../../3-networks/envs/development",
+		},
+		{
+			name:            "bu2_non-production",
+			sharedData:      "bu2_shared",
+			perimeterEnvVar: "TF_VAR_nonprod_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_2/non-production",
+			networkTfDir:    "../../../3-networks/envs/non-production",
+		},
+		{
+			name:            "bu2_production",
+			sharedData:      "bu2_shared",
+			perimeterEnvVar: "TF_VAR_prod_restricted_service_perimeter_name",
+			tfDir:           "../../../4-projects/business_unit_2/production",
+			networkTfDir:    "../../../3-networks/envs/production",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+
+			netVars := map[string]interface{}{
+				"access_context_manager_policy_id": policyID,
+			}
+			networks := tft.NewTFBlueprintTest(t,
```

Comment:
Lets add a comment that we create this inorder to retrieve output.

Reply:
comment added